### DISCUSSION TITLE
Enforcement of cascaded failures in CI

### DISCUSSION THREAD

**Discussion by BillyONeal** [microsoft/vcpkg#20180] [Ideas]
This is a continued discussion from https://github.com/microsoft/vcpkg/pull/20074#issuecomment-916565430

Pending things:

- [ ] Fast tracked review of @dg0yt's PR #19259
- [ ] Fix qt in a less correct way than described in #19922 (it will waste disk space) but which will restore the unique port attribution rule ( @ras0219 / @ras0219-msft ) https://github.com/microsoft/vcpkg/pull/20145
- [x] Logging of the count of "cascades at the top" -- those that result from ci.baseline.txt and "supports" -- and keep track of this as an important metric in CI. https://github.com/microsoft/vcpkg-tool/pull/188
- [x] ~~Add validation for the preceding number and fail PRs if the number of cascades change. Then at least git blame will find PRs that had to make significant edits here.~~ https://github.com/microsoft/vcpkg/pull/20140
- [ ] Add `=cascade` or similar to ci.baseline.txt. This probably requires moving parsing of that file into vcpkg.exe rather than a preprocessing script. (Probably @BillyONeal but not on the top of priority list for now)

Experience notes:

Unfortunately we were forced to disable the cascade count check because:
* it caused stealth merge conflicts with master
  * any time 2 PRs are merged that change a supports or ci.baseline.txt status
  * any time 1 PR is merged that add a new port with supports or ci.baseline.txt change and another PR is merged with one such change
* the way we calculate skips is different in the nightly CI run (we skip less stuff in CI) meaning nightly CIs were always doomed with that enforcement turned on.

Backing that out is https://github.com/microsoft/vcpkg/pull/20181

We still think the =cascade plan can work but we are hoping there are better answers here...

/cc @dg0yt @cenit

**Comment by wrobelda** [comment]
PS. This will also effectively provide a fix for https://github.com/microsoft/vcpkg/issues/13533 when implemented.
